Waypoint[] waypoints
string[] agent_list
